### PR TITLE
[AMORO-2996] Reload catalogs page after removing the existing catalog

### LINKED ISSUE #2996: [Improvement]: Refresh the catalog list after remove one catalog
### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/amoro/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Refresh the catalog list after remove one catalog.

image

### How should we improve?

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Subtasks

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by tcodehuber**
## Why are the changes needed?

Close #2996.

## Brief change log

-

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: amoro-ams/amoro-ams-dashboard/src/views/catalogs/Detail.vue [hunk 1]
```diff
@@ -533,9 +535,15 @@ async function deleteCatalogModal() {
   Modal.confirm({
     title: t('deleteCatalogModalTitle'),
     onOk: async () => {
-      await delCatalog(formState.catalog.name || '')
-      message.success(`${t('remove')} ${t('success')}`)
-      emit('updateEdit', false)
+      try {
+        await delCatalog(formState.catalog.name || '')
+        message.success(`${t('remove')} ${t('success')}`, 1, () => {
+          router.replace({ path: '/catalogs', query: {} }).then(() => router.go(0))
```
Comment (chouchouji) [inline amoro-ams/amoro-ams-dashboard/src/views/catalogs/Detail.vue:541]: Is this operation to refresh the page?

Comment (tcodehuber) [reply amoro-ams/amoro-ams-dashboard/src/views/catalogs/Detail.vue:541]: > Is this operation to refresh the page?

Yes

Comment (chouchouji) [reply amoro-ams/amoro-ams-dashboard/src/views/catalogs/Detail.vue:541]: Calling [getConfigInfo](https://github.com/apache/amoro/blob/master/amoro-ams/amoro-ams-dashboard/src/views/catalogs/Detail.vue#L233C16-L233C29) function again maybe also refresh the data. Can you have a try?

Comment (chouchouji) [reply amoro-ams/amoro-ams-dashboard/src/views/catalogs/Detail.vue:541]: I guess the catalog list should be refreshed after deleting a catalog, so is there an event (like delete event) to tell the `index.vue` component to refresh the list? If my inference is wrong, please correct me. Thanks!

**Review by chouchouji — APPROVED** [review]

**Review by zhoujinsong — APPROVED** [review]
LGTM.

Thanks for the contribution, @tcodehuber 
Thanks for the review, @chouchouji
